crates/common/mqtt_channel/src/config.rs

```diff
@@ -48,6 +51,34 @@ pub struct Config {
     ///
     /// Default: None
     pub last_will_message: Option<Message>,
+
+    /// With first message on connection
+    ///
+    /// Default: None
+    pub initial_message: Option<InitMessageFn>,
+}
+
+#[derive(Clone)]
+pub struct InitMessageFn {
+    initfn: Arc<Box<dyn Fn() -> Message + Send + Sync>>,
+}
+
+impl InitMessageFn {
+    pub fn new(call_back: impl Fn() -> Message + Sync + Send + 'static) -> InitMessageFn {
+        InitMessageFn {
+            initfn: Arc::new(Box::new(call_back) as Box<dyn Fn() -> Message + Sync + Send>),
+        }
+    }
+
+    pub fn call(&self) -> Message {
```

Comment:
Better to rename this function.

The name `call` fails to convey the intent (creating a new message) and leaks internal details (use of a closure).


```suggestion
    pub fn new_init_message(&self) -> Message {
```